src/frontend/running.rs

```diff
@@ -287,13 +291,22 @@ impl RunningView {
                             backend; qed",
                         ),
                         FarmWidgetInit {
-                            farm,
+                            farm: farm.clone(),
```

Comment:
Instead of cloning the whole data structure you could have incremented space pledged first, then cloning would not be necessary.